### PR TITLE
13 account modal wizard

### PR DESCRIPTION
**Pull request by cdcabrera**
### What's in this PR
* wizard steps styling and copy updates
* rename account modal step three
* wizard redux to include ARN step
* tooltip-popover styling
* temp service added for addAccount
* unit test snapshots update, wizard arn step testing

### What's NOT in this PR
- ~form field validation for the ARN, regex pattern still required~ basic regex prefix check added
- ~API endpoint for the ARN post... a mock endpoint was created instead to complete the work.~ 
  - This can be tested using the command `$ npm install; npm start` on a local fork/clone/branch of the repo

### Updated Demos

jul-05-2018 22-53-07

jul-05-2018 22-49-16

### Prior Demos

jul-03-2018 03-42-08

#### Pending state after submitting the new account
screen shot 2018-07-03 at 1 15 18 pm

##### Pending cancel
jul-03-2018 13-16-29

#### "Current" Error state on API error
screen shot 2018-07-03 at 1 12 38 pm

updates #13

### CODE REVIEW

**Comment by elyezer** [comment]
> form field validation for the ARN, regex pattern still required

Is there an issue created for this already?

> API endpoint for the ARN post... a mock endpoint was created instead to complete the work.

Same here.

Let's make sure we have open issues for those so we can get this merged and move on, what do you think?

**Comment by elyezer** [comment]
I was trying to verify the related issues AC and found that when trying to create with a valid ARN the wizard gives an error and I see that some information is missing from the POST request:

```
resourcetype: "This field is required"
```

After getting the above error I closed the wizard and started it again. Going to the second step _Role_ the Cloudigrade account id text field is empty and clicking on copy does not copy anything. Check the screen shot below:

screenshot from 2018-07-04 12-11-07

Is the above issues something that this PR should take care of? I can't verify more items because some of them are TBD and getting the error I can't validate that the cloud account was actually saved.

**Comment by cdcabrera** [comment]
@elyezer can do on the opening of issues for the ARN regex and API endpoint.

Step 2 however, based on the screenshot, looks like it didn't fire the API call again, taking a look, good catch!

**Comment by cdcabrera** [comment]
@elyezer @kdelee  regarding Step 2 not loading the API call for the account ID. There looks like there may be inconsistent behavior around how the API responds to the UI, had trouble telling whether that was a testing environment issue or an actual problem. 

Reason for not thinking its the UI service call:
* From the UIs perspective we make the API call when the wizard opens, looking at the dev tools inspector. The API call fires, but it appears as an empty response under the guise of a 200/201 level success. If you all are able to confirm that behavior that would be fantastic

**Comment by kdelee** [comment]
Woop! 

screenshot from 2018-07-06 13-26-16


and confirmed it is on the server with api client from integrade:
```
In [2]: from integrade import api

In [3]: client = api.Client()

In [4]: client.get('/api/v1/account/')
Out[4]: <Response [200]>

In [5]: client.get('/api/v1/account/').json()
Out[5]: 
{'count': 1,
 'next': None,
 'previous': None,
 'results': [{'account_arn': 'arn:aws:iam::311230538223:role/role-for-cloudigrade',
   'aws_account_id': '311230538223',
   'created_at': '2018-07-06T17:26:12.249630Z',
   'id': 8,
   'name': 'my dev08 account',
   'updated_at': '2018-07-06T17:26:12.249661Z',
   'url': 'http://test.cloudigra.de/api/v1/account/8/',
   'user_id': 16,
   'resourcetype': 'AwsAccount'}]}
```

Only thing that was a bit not-pretty was text in 502 error. Granted we need to figure out how to get rid of all these 502/503/504 errors but this is not so good looking:
screenshot from 2018-07-06 13-24-51


You could write and issue and follow up with 500 error at another time because I think there is other error handling decisions to be made.

**Review by kdelee — DISMISSED** [review]
:ship: :it:

**Comment by cdcabrera** [comment]
awesome @kdelee ! and can do on placing a nicer 500 level check in there as a follow up

File: src/components/accountWizard/accountWizardStepResults.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { Spinner } from 'patternfly-react';
+import { connect } from '../../redux';
+import apiTypes from '../../constants/apiConstants';
+
+const AccountWizardStepResults = ({ accountName, edit, error, errorMessage, fulfilled, pending }) => (
+  <React.Fragment>
+    {error && (
+      <div className="wizard-pf-complete blank-slate-pf">
+        <div className="wizard-pf-success-icon">
+          <span className="pficon pficon-error-circle-o" />
```
Comment (jeff-phillips-18) [inline src/components/accountWizard/accountWizardStepResults.js]: Could use the pf-react `Icon` component here. It would add the aria-hidden tag.

**Review by jeff-phillips-18 — DISMISSED** [review]
One small nit, LGTM.

File: src/components/accountWizard/accountWizardStepResults.js (hunk 1 above)
Comment (cdcabrera) [reply src/components/accountWizard/accountWizardStepResults.js]: Can do!

**Review by jeff-phillips-18 — APPROVED** [review]
LGTM

**Comment by elyezer** [comment]
> awesome @kdelee ! and can do on placing a nicer 500 level check in there as a follow up

@cdcabrera I can haz an issue?

**Comment by elyezer** [comment]
Maybe that can be handled when working on https://github.com/cloudigrade/frontigrade/issues/43. If having a better 5XX handler does not make sense to be on that issue feel free to remove the bits related to it on the issue.

**Pull request merged**